syntax = "proto2";
package tutorial;


option go_package = "gateway/proto";

enum DeviceType {
    DEVICE_TYPE_ACTUATOR = 0;
    DEVICE_TYPE_SENSOR = 1;
}

message ConnectionRequest {
    optional string queue_name = 1;
    optional DeviceType type = 2;
    optional string ip = 3;
    optional string port = 4;
    optional string data = 5;
}

message ConnectionResponse {
    optional string id = 1;
}

message DisconnectionRequest {
    optional string queue_name = 1;
    optional string id = 2;
}

message SensorDataUpdate {
    optional string data = 1;
    optional string id = 2;
    optional string name = 3;
}
